src/resources/parser/glb-container-resource.js

```diff
@@ -190,6 +192,50 @@ class GlbContainerResource {
         return GlbContainerResource.createSceneHierarchy(sceneClones, 'Entity');
     }
 
+    // get material variants
+    getMaterialVariants() {
+        return Object.keys(this.data.variants);
+    }
+
+    // apply material variant to entity
+    applyMaterialVariant(entity, name) {
+        const variant = name ? this.data.variants[name] : null;
+        if (variant === undefined) {
+            Debug.warn(`No variant named ${name} exists in resource`);
+            return;
+        }
+        const renders = entity.findComponents("render");
+        for (let i = 0; i < renders.length; i++) {
+            const renderComponent = renders[i];
+            this._applyMaterialVariant(variant, renderComponent.meshInstances);
+        }
+    }
+
+    // apply material variant to mesh instances
+    applyMaterialVariantInstances(instances, name) {
+        const variant = name ? this.data.variants[name] : null;
+        if (variant === undefined) {
+            Debug.warn(`No variant named ${name} exists in resource`);
+            return;
+        }
+        this._applyMaterialVariant(variant, instances);
+    }
+
+    // internally apply variant to instances
+    _applyMaterialVariant(variant, instances) {
+        instances.forEach((instance) => {
+            if (variant === null) {
+                instance.material = this._defaultMaterial;
+            } else {
+                const meshVariants = this.data.meshVariants[instance.mesh.id];
+                if (meshVariants) {
+                    instance.material = this.data.materials[meshVariants[variant]];
+                    Debug.assert(instance.material);
```

Comment:
if you could move the assert outside of if/else, so that it validates both cases.